clang/test/CheckedC/inferred-bounds/widened-bounds-check.c

```diff
@@ -13,7 +13,7 @@ void f1() {
   {}
 
   if (*p) {
-    p++;
+    p++; // expected-error {{declared bounds for p are invalid after assignment}}
```

Comment:
This message could confuse programmers because they might think you are referring to an assignment operator.   There is no assignment operator in that expression.   You might want a message that says increment or decrement instead.  It is fine to address this in a follow-up PR.

Reply:
Created issue #844 to track this.